syntax = "proto3";

package skill_manager;

message Action {
    string file = 1;		// HDF5格式动作文件
    int32 fps = 2;		// 回放帧率
    int32 timeout_ms = 3;	// 超时时间
}

message Audio {
    string file = 1;		// WAV格式音频文件
    int32 timeout_ms = 2;	// 超时时间
}

message Response {
    bool succeeded = 1;
    string msg = 2;
}

message ActionResponse {
    bool succeeded = 1;
    string msg = 2;
    int32 precentage = 3;	// 动作进度百分比
}

service SkillManager {
    rpc startAction(Action) returns (stream ActionResponse);
    rpc stopAction(Action) returns (Response);
    rpc playAudio(Audio) returns (Response);
    rpc stopAudio(Audio) returns (Response);
}